extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/contacts/GoogleContactsExporter.java

```diff
@@ -116,7 +116,7 @@ private static void convertToVCardNamesAndPopulate(VCard vCard, List<Name> perso
     for (Name personName : personNames) {
       StructuredName structuredName = convertToVCardNameSingle(personName);
       Boolean isNamePrimary = personName.getMetadata().getPrimary();
-      if (isNamePrimary != null && isNamePrimary) {
+      if (primaryStructuredName == null && isNamePrimary != null && isNamePrimary) {
```

Comment:
Could you update the comment on line 116 to state how we handle this case?

Reply:
Oops, I meant to update the existing comment on like 113 since its not longer applicable

Reply:
Thanks @seehamrun, Can you please check now?